clusterloader2/run-e2e.sh

```diff
@@ -29,14 +29,27 @@ export AZUREDISK_CSI_DRIVER_VERSION="v1.15.0"
 export AZUREDISK_CSI_DRIVER_INSTALL_URL="https://raw.githubusercontent.com/kubernetes-sigs/azuredisk-csi-driver/${AZUREDISK_CSI_DRIVER_VERSION}/deploy/install-driver.sh"
 export WINDOWS_USE_HOST_PROCESS_CONTAINERS=true
 
-# Deploy the GCP PD CSI Driver if required
+# Deploy the GCE PD CSI Driver if required
 if [[ "${DEPLOY_GCI_DRIVER:-false}" == "true" ]]; then
    if [[ -z "${E2E_GOOGLE_APPLICATION_CREDENTIALS:-}" ]]; then
-      echo "Env var E2E_GOOGLE_APPLICATION_CREDENTIALS must be set to deploy driver"
-      exit 1
+      # Running yq to patch GCE PD CSI Driver manifests so that it runs using
+      # Application Default Credentials instead of creating service account
+      # keys, hence avoiding security risks. See
+      # https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/pull/610
+      # for more details.
+      yq eval -i '
+         with(select(.metadata.name == "csi-gce-pd-controller" and .kind == "Deployment").spec.template.spec;
+            del(.volumes[] | select(.name == "cloud-sa-volume")) |
+            with(.containers[] | select(.name == "gce-pd-driver");
+               del(.env[] | select(.name == "GOOGLE_APPLICATION_CREDENTIALS")) |
+               del(.volumeMounts[] | select(.name == "cloud-sa-volume"))
+            )
+         )' "${CLUSTERLOADER_ROOT}"/drivers/gcp-csi-driver-stable.yaml
```

Comment:
It should create a copy, not modify in place. In subsequent run, CL2 would behave differently with in-place modification.

Reply:
Good point; done.